### PR TITLE
Client documentation and docker image

### PR DESCRIPTION
**Pull request by niccokunzmann**
I built a client docker image and documented how to setup the client yourself.

A current client-broker-conversation looks like this: https://github.com/wlanslovenija/tunneldigger/pull/99#issuecomment-507455889

### CODE REVIEW

**Comment by niccokunzmann** [comment]
With this fixed IP-Address, the broker can reach the client.

```
root@broker:/opt/src# ping 10.254.0.3
PING 10.254.0.3 (10.254.0.3) 56(84) bytes of data.
64 bytes from 10.254.0.3: icmp_seq=1 ttl=64 time=0.061 ms
64 bytes from 10.254.0.3: icmp_seq=2 ttl=64 time=0.081 ms
64 bytes from 10.254.0.3: icmp_seq=3 ttl=64 time=0.078 ms
64 bytes from 10.254.0.3: icmp_seq=4 ttl=64 time=0.080 ms
64 bytes from 10.254.0.3: icmp_seq=5 ttl=64 time=0.066 ms
64 bytes from 10.254.0.3: icmp_seq=6 ttl=64 time=0.077 ms
64 bytes from 10.254.0.3: icmp_seq=7 ttl=64 time=0.080 ms

```

File: client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+COPY CMakeLists.txt /srv/tunneldigger/client/CMakeLists.txt
+COPY l2tp_client.c /srv/tunneldigger/client/l2tp_client.c
+COPY libasyncns /srv/tunneldigger/client/libasyncns
+
+RUN apt-get update && \
+    apt-get install -y \
+        cmake build-essential iproute2 pkg-config curl bison flex && \
+    mkdir -p /srv/libnl && \
+    cd /srv/libnl && \
+    curl -sL https://github.com/thom311/libnl/releases/download/libnl3_4_0/libnl-3.4.0.tar.gz \
```
Comment (RalfJung) [inline client/Dockerfile]: This is odd, I can build the client with just system packages on my Debain box. Is that not possible with Ubuntu?

I have installed `libnl-3-dev` and `libnl-genl-3-dev`, not sure which of them is needed.

File: docs/client.rst [hunk 2]
```diff
@@ -1,17 +1,100 @@
...
+
+There is a docker image available from `wlanslovenija/tunneldigger-client
+<https://hub.docker.com/r/wlanslovenija/tunneldigger-client>`_.
+You can run it when you installed docker.
+
+.. code:: shell
+
+    docker run --rm -it --privileged wlanslovenija/tunneldigger-client -b <IP>:<PORT>
+
+Replace ``<IP>`` and ``<PORT>`` with the ip and port the broker uses.
```
Comment (RalfJung) [inline docs/client.rst]: I guess my main question here is "why" -- when is this docker image actually *useful*?

But I am fine if the answer is just "to document a reproducible way to build and run the client".

File: client/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    cd /srv/tunneldigger/client/ && \
+    cmake . && \
+    make && \
+    mv tunneldigger /usr/local/bin/tunneldigger && \
+    apt-get -y purge cmake build-essential pkg-config curl && \
+    apt-get -y autoremove
+
+COPY docker/* /srv/tunneldigger/client/
+
+ENTRYPOINT ["/srv/tunneldigger/client/run.sh"]
```
Comment (RalfJung) [inline client/Dockerfile]: Would it make sense to add a CI job that builds this image? Just to make sure it keeps working?

Or is that subsumed by your server-side docker PR?

File: client/docker/run.sh [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    i=$((i+1))
+    shift
+  fi
+done
+
+#
+# Run tunneldigger
+#
+echo /usr/local/bin/tunneldigger -f -i "$IFACE" -s "$HOOK" -u "$UUID" "${args[@]}"
+/usr/local/bin/tunneldigger -f -i "$IFACE" -s "$HOOK" -u "$UUID" "${args[@]}"
```
Comment (RalfJung) [inline client/docker/run.sh]: Would it make sense to `exec /usr/local/bin/tunneldigger ...`? This is just a wrapper to start the "main" process after all.

**Comment by RalfJung** [comment]
Closing due to lack of activity. If you want to pick this up again @niccokunzmann, feel free to reopen!

**Pull request closed without merging**